queue_job/data/queue_data.xml

```diff
@@ -1,17 +1,6 @@
 <?xml version="1.0" encoding="utf-8" ?>
 <odoo>
     <data noupdate="1">
-        <record id="ir_cron_queue_job_garbage_collector" model="ir.cron">
```

Comment:
I have archived the cron in the `pre-migration.py`. Therefore, there won't be any error.
I always prefer to archive (set active to false) rather than deleting.